src/unix/pty.cc

```diff
@@ -663,25 +677,13 @@ pty_getproc(int fd, char *tty) {
 #elif defined(__APPLE__)
 
 static char *
-pty_getproc(int fd, char *tty) {
-  int mib[4] = { CTL_KERN, KERN_PROC, KERN_PROC_PID, 0 };
-  size_t size;
-  struct kinfo_proc kp;
-
-  if ((mib[3] = tcgetpgrp(fd)) == -1) {
-    return NULL;
-  }
-
-  size = sizeof kp;
-  if (sysctl(mib, 4, &kp, &size, NULL, 0) == -1) {
-    return NULL;
-  }
-
-  if (size != (sizeof kp) || *kp.kp_proc.p_comm == '\0') {
+pty_getproc(int pid) {
+  char pname[MAXCOMLEN + 1];
+  if (!proc_name(pid, pname, sizeof(pname))) {
     return NULL;
   }
 
-  return strdup(kp.kp_proc.p_comm);
+  return strdup(pname);
```

Comment:
I think this is similar to tmux's technique, are we going to regress anywhere by doing it this way? I don't understand how it worked before.

Reply:
I went based on the implementation of `proc_name` which is used in much widely across the kernel code https://github.com/apple/darwin-xnu/blob/2ff845c2e033bd0ff64b5b6aa6063a1f8f65aa32/bsd/kern/kern_proc.c#L1029-L1044 and it seems reliable.